diff --git a/test/assignment1/username-from-conf-file.h b/test/assignment1/username-from-conf-file.h
index c83a06b..7e286f7 100644
--- a/test/assignment1/username-from-conf-file.h
+++ b/test/assignment1/username-from-conf-file.h
@@ -8,7 +8,7 @@
  */
 static inline char *malloc_username_from_conf_file()
 {
-    size_t len = 0;
+    size_t len = 1;
     // Note: this buffer will be reallocated in getline() as necessary
     char *buffer = malloc(len + 1);
     buffer[0] = '\0';
@@ -38,6 +38,7 @@ static inline char *malloc_username_from_conf_file()
         } else {
             printf("Read %s from conf/username.txt\n", buffer);
         }
+        fclose(fp);
     } else {
         fprintf(stderr, "Could not open conf/username.txt for reading\n");
     }
